### PR TITLE
Add `cleanup_instance_files` step to clean up data of stopped instance.

### LINKED ISSUE #335: [4pt] Add a step to cleanup instance data
It can be used to write playbook to re-bootstrap a replica.

### PR DESCRIPTION
**Pull request by hackallcode**
Closes #335

Before the patch, it was impossible to write playbook to re-bootstrap a replica,
because there was no step to clear the instance data.

In this patch added `cleanup_instance_files` step to clean up data of stopped instance.

### CODE REVIEW

File: doc/scenario.md [hunk 1]
```diff
@@ -852,3 +853,17 @@ Input facts (set by config):
 
 - `cartridge_runtime_params` - new instance parameters ([more details here](/doc/instances.md));
 - `expelled` - indicates if instance must be expelled from topology.
+
+### cleanup_instance
```
Comment (dokshina) [inline doc/scenario.md]: `clanup_expelled = stop_service + cleanup_instance_files`
`cleanup_instance =  cleanup_instance_files`


Maybe `cleanup_instance_files` is a better name?

File: doc/scenario.md [hunk 2]
```diff
@@ -852,3 +853,17 @@ Input facts (set by config):
...
+Clean up data of stopped instance.
+If instance is started, an error will be returned.
+
+Input facts (set by role):
+
+- `instance_info` - information for a current instance ([more details here](#role-facts-descriptions)).
+
+Input facts (set by config):
+
+- `cartridge_cleanup_excluded_paths` - list of folders and files
```
Comment (dokshina) [inline doc/scenario.md]: `cartridge_keep_files_on_cleanup`?

File: library/cartridge_check_instance_not_started.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from ansible.module_utils.helpers import Helpers as helpers
+
+argument_spec = {
+    'console_sock': {'required': True, 'type': 'str'},
+}
+
+
+def check_state(params):
+    if helpers.is_instance_started(params['console_sock']):
+        return helpers.ModuleRes(failed=True, msg='instance is not stopped')
```
Comment (dokshina) [inline library/cartridge_check_instance_not_started.py]: What if the instance wasn't started? Maybe `msg='Instance is running'` is better?

File: tasks/steps/cleanup_instance.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+
+- name: 'Check that instance is not started'
```
Comment (dokshina) [inline tasks/steps/cleanup_instance.yml]: msg='instance is not running'

File: tasks/steps/cleanup_instance.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    console_sock: '{{ instance_info.console_sock }}'
+  register: check_instance_not_started_res
+
+- name: 'Cleanup instance files'
+  file:
+    path: '{{ file_path }}'
+    state: absent
+  loop_control:
+    loop_var: file_path
+  with_items: '{{ instance_info.files_to_remove_on_cleanup }}'
```
Comment (dokshina) [inline tasks/steps/cleanup_instance.yml]: `cartridge_cleanup_excluded_paths` isn't applied here.

File: doc/scenario.md (hunk 2 above)
Comment (hackallcode) [reply doc/scenario.md]: cartridge_paths_to_keep_on_cleanup

File: tasks/steps/cleanup_instance.yml (hunk 5 above)
Comment (hackallcode) [reply tasks/steps/cleanup_instance.yml]: I'm not sure that it's good to allow the user to ignore PID and socket

**Review by dokshina — APPROVED** [review]

**Pull request merged**